test/integration/transaction.test.js

```diff
@@ -527,6 +527,93 @@ if (current.dialect.supports.transactions) {
 
```

Comment:
New tests are creating [nested promises](https://medium.com/@pyrolistical/how-to-get-out-of-promise-hell-8c20e0ab0513), please improve that code where you can

Reply:
Thanks, this looked like the style other tests in the file were using to me, but I'm happy to adjust it